docs/reference_guides/model_libraries/generic/costing/process_costing.rst

```diff
@@ -0,0 +1,865 @@
+
+.. note:: number of units by default is fixed to 1 and the user must unfix this variable to optimize the number of units. Also, `number of units` can be built as a continuous variable or an integer variable. If the latter, the user must provide an mip solver. Use the global costing argument for this purpose (integer_n_units=True or False).
+
+Example
+^^^^^^^
+Below is a simple example of how to add cost correlations to a flowsheet including a heat exchanger using the default IDAES costing module.
+
+
+.. code:: python
+
+    from pyomo.environ import ConcreteModel, SolverFactory
+    from pyomo.util.calc_var_value import calculate_variable_from_constraint
+    from idaes.core import FlowsheetBlock
+    from idaes.models.unit_models.heat_exchanger import HeatExchanger, HeatExchangerFlowPattern
+    from idaes.models.properties import iapws95
+    from idaes.core.util.model_statistics import degrees_of_freedom
+
+    from idaes.models.costing.SSLW import SSLWCosting, SSLWCostingData
+    from idaes.core import UnitModelCostingBlock
+    from idaes.models.costing.SSLW import HeaterMaterial, HeaterSource
+    
+    m = ConcreteModel()
+    m.fs = FlowsheetBlock(default={"dynamic": False})
+    
+    m.fs.properties = iapws95.Iapws95ParameterBlock()
+    
+    m.fs.unit = HeatExchanger(default={
+                "shell": {"property_package": m.fs.properties},
+                "tube": {"property_package": m.fs.properties},
+                "flow_pattern": HeatExchangerFlowPattern.countercurrent})
+    # set inputs
+    m.fs.unit.shell_inlet.flow_mol[0].fix(100)     # mol/s
+    m.fs.unit.shell_inlet.enth_mol[0].fix(3500)    # j/s
+    m.fs.unit.shell_inlet.pressure[0].fix(101325)  # Pa 
+    
+    m.fs.unit.tube_inlet.flow_mol[0].fix(100)
+    m.fs.unit.tube_inlet.enth_mol[0].fix(4000)
+    m.fs.unit.tube_inlet.pressure[0].fix(101325.0)
+    
+    m.fs.unit.area.fix(1000)  # m2
+    m.fs.unit.overall_heat_transfer_coefficient.fix(100)  # W/m2K
+    
+    m.fs.costing = SSLWCosting()
+    m.fs.H101.costing = UnitModelCostingBlock(
+        default={
+            "flowsheet_costing_block": m.fs.costing,
+            "costing_method": SSLWCostingData.cost_fired_heater,
+            "costing_method_arguments": {
+                    "material_type": HeaterMaterial.CarbonSteel,
+                    "heat_source": HeaterSource.Fuel
+            }
+        }
+    )
+
+    m.fs.unit.initialize()
+
+    opt = SolverFactory('ipopt')
+    opt.options = {'tol': 1e-6, 'max_iter': 50}
+    results = opt.solve(m, tee=True)
+
+Units
+^^^^^
+
+It is important to highlight that the costing method interrogates the property 
+package to determine the units of this model, if the user provided the correct 
+units in the metadata dictionary (see property models for additional information), 
+the model units will be converted to the right units. 
+For example: in this example area is in m^2, while the cost correlations for heat 
```

Comment:
Formatting: "m^2" should be rendered in math mode to get a proper superscript. I've seen at least one other example of this later in the file as well.